src/amo/components/UserProfile/index.js

```diff
@@ -0,0 +1,170 @@
+
+    if (!user) {
+      dispatch(fetchUserAccount({
+        errorHandlerId: errorHandler.id,
+        username: params.username,
+      }));
+    }
+  }
+
+  componentWillReceiveProps({ params: newParams }: { params: UserProfileParams }) {
+    const { dispatch, errorHandler, params: oldParams } = this.props;
+
+    if (oldParams.username !== newParams.username) {
+      dispatch(fetchUserAccount({
+        errorHandlerId: errorHandler.id,
+        username: newParams.username,
+      }));
+    }
+  }
+
+  render() {
+    const {
+      errorHandler,
+      i18n,
+      params,
+      user,
+    } = this.props;
+
+    if (errorHandler.hasError()) {
+      log.warn('Captured API Error:', errorHandler.capturedError);
+
+      if (errorHandler.capturedError.responseStatusCode === 404) {
+        return <NotFound errorCode={errorHandler.capturedError.code} />;
+      }
+    }
+
+    const userProfileHeader = (
+      <React.Fragment>
+        <UserAvatar className="UserProfile-avatar" user={user} />
+
+        <h1 className="UserProfile-name">
+          {user ? user.displayName : <LoadingText />}
+        </h1>
+      </React.Fragment>
+    );
+    const userProfileTitle = i18n.sprintf(
+      i18n.gettext('User Profile for %(user)s'), {
+        user: user ? user.displayName : params.username,
+      }
+    );
+
+    return (
+      <div className="UserProfile">
+        <Helmet>
+          <title>{userProfileTitle}</title>
+        </Helmet>
+
+        {errorHandler.renderErrorIfPresent()}
+
+        <div className="UserProfile-wrapper">
+          <Card
+            className="UserProfile-user-info"
+            header={userProfileHeader}
+          >
+            <DefinitionList className="UserProfile-dl">
+              {user && user.homepage ? (
+                <Definition
+                  className="UserProfile-homepage"
+                  term={i18n.gettext('Homepage')}
+                >
+                  {/* TODO: Make a helper that strips https?:\/\/ from text */}
+                  <a href={user.homepage}>{user.homepage.replace(/^https?:\/\//, '')}</a>
+                </Definition>
+              ) : null}
+              <Definition
+                className="UserProfile-user-since"
+                term={i18n.gettext('User since')}
+              >
+                {user ? i18n.moment(user.created).format('ll')
+                      : <LoadingText />}
+              </Definition>
+              <Definition
+                className="UserProfile-number-of-addons"
+                term={i18n.gettext('Number of add-ons')}
+              >
+                {user ? user.num_addons_listed : <LoadingText />}
+              </Definition>
+              <Definition
+                className="UserProfile-rating-average"
+                term={i18n.gettext('Average rating of developer’s add-ons')}
+              >
+                {user ? (
+                  <Rating
+                    rating={parseFloat(user.average_addon_rating)}
```

Comment:
Why did you need `parseFloat()`? `Rating` already [calls it](https://github.com/mozilla/addons-frontend/blob/master/src/ui/components/Rating/index.js#L98).

Reply:
Ohhhhhhh, I think it's because of how things are called. `Rating` calls `toFixed(1)` first, but I think the data returned by the API caused an issue with `.toFixed(1)`. Maybe `Rating` needs to be fixed.

Reply:
Ah, and it's because of https://github.com/mozilla/addons-server/issues/7822

I can either add the `parseFloat` to the user reducer or here until https://github.com/mozilla/addons-server/issues/7822 is fixed, or keep the fix to `Rating` I added.